universitetet-i-oslo-rettsvitenskap.csl

```diff
@@ -0,0 +1,295 @@
+  </citation>
+  <bibliography entry-spacing="0">
+    <sort>
+      <key macro="type-sorting"/>
+      <key macro="author-full"/>
+      <key variable="issued"/>
+      <key variable="title"/>
+    </sort>
+    <layout>
+      <choose>
+        <if type="article-journal">
+          <group suffix=".">
+            <text macro="author-full"/>
+            <text prefix=". " macro="title"/>
+            <text prefix=", " font-style="italic" variable="container-title"/>
+            <text prefix=" Årg. " variable="volume"/>
+            <text prefix=" " macro="issued"/>
+            <text prefix=", s. " variable="page"/>
+            <choose>
+              <if variable="DOI" match="none">
+                <text prefix=", " variable="URL"/>
+              </if>
+              <else>
+                <text prefix=", " variable="DOI"/>
+              </else>
+            </choose>
+            <text prefix=" " macro="retrieved-from"/>
+          </group>
+        </if>
+        <else-if type="article-newspaper article-magazine" match="any">
+          <group suffix=".">
+            <text macro="author-full"/>
+            <text prefix=". " macro="issued"/>
+            <text prefix=" " macro="title"/>
+            <text prefix=", " font-style="italic" variable="container-title"/>
+            <choose>
+              <if variable="URL" match="none">
+                <text prefix=", " macro="issued-full-date"/>
+              </if>
+              <else>
+                <text prefix=", " variable="URL"/>
+                <text macro="accessed-date"/>
+              </else>
+            </choose>
+          </group>
+        </else-if>
+        <else-if type="book thesis" match="any">
+          <group suffix=".">
+            <text macro="author-full"/>
+            <text prefix=". " macro="title"/>
+            <text prefix=", " suffix=" utg." variable="edition"/>
+            <text prefix=", " suffix=":" variable="publisher-place"/>
+            <text prefix=" " variable="publisher"/>
+            <text prefix=", " macro="issued-no-parenthesis"/>
+          </group>
+        </else-if>
+        <else-if type="chapter paper-conference" match="any">
+          <group suffix=".">
+            <text macro="author-full"/>
+            <text prefix=". " macro="title"/>
+            <text prefix=", i " font-style="italic" variable="container-title"/>
+            <text prefix=", " suffix=" (red.)" macro="editor"/>
+            <text prefix=", " suffix=". utg.," variable="edition"/>
+            <text prefix=", " suffix=": " variable="publisher-place"/>
+            <text variable="publisher"/>
+            <text prefix=", " macro="issued-no-parenthesis"/>
+            <text prefix=", " variable="URL"/>
+            <text prefix=" " macro="accessed-date"/>
+          </group>
+        </else-if>
+        <else-if type="legislation bill" match="any">
+          <group suffix=".">
+            <text suffix="            " macro="issued-no-parenthesis"/>
```

Comment:
@hgsl, what are you trying to achieve here with the `suffix="            "`? I'm wondering if there isn't a more elegant way.

Reply:
This is actually done intentionally and is in line with our style. 

The purpose is, for certain reference types such as legislation, court cases, and certain other types of legal sources, to have them in separate registers apart from the literature. In these registers the bibliographic information is separated into two seperate columns, for ease of reference. This is also why I have incorporated the type sorting, so that it is easier for the user to separate the registers from the main bibliography.

You can see examples of this in the law library's specifications: https://www.ub.uio.no/skrive-publisere/referere/fag/rettsvitenskap.html

E.g. Norske rettskilder > Lover og forskrifter

Reply:
That won't work well with spaces, though, a) because they'll align unevenly where dates aren't 4 characters (e.g. because of disambiguation) and b) because newlines will start at the front of the line. You should instead use `display` properties as explained here: http://docs.citationstyles.org/en/stable/specification.html#display

Reply:
Yep, you're definitely right about that. Thanks for the tip, I'll work on it some more when I have the time. Again, I really appreciate the feedback!